src/de/mod.rs

```diff
@@ -227,7 +227,28 @@ impl<'de, 'a, R: Read> de::Deserializer<'de> for &'a mut Deserializer<R> {
     deserialize_type!(deserialize_u64 => visit_u64);
     deserialize_type!(deserialize_f32 => visit_f32);
     deserialize_type!(deserialize_f64 => visit_f64);
-    deserialize_type!(deserialize_bool => visit_bool);
+
+    fn deserialize_bool<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
+        if let XmlEvent::StartElement { .. } = *self.peek()? {
+            self.set_map_value()
+        }
+        self.read_inner_value::<V, V::Value, _>(|this| {
+            if let XmlEvent::EndElement { .. } = *this.peek()? {
+                return visitor.visit_bool(false);
+            }
+            expect!(this.next()?, XmlEvent::Characters(s) => {
+                match s.as_str() {
+                    "true" | "1" => visitor.visit_bool(true),
+                    "false" | "0" => visitor.visit_bool(false),
+                    _ => Err(ErrorKind::UnexpectedToken(
```

Comment:
Maybe better to use [`Error::unknown_variant`](https://docs.serde.rs/serde/de/trait.Error.html#method.unknown_variant) or [`Error::invalid_value`](https://docs.serde.rs/serde/de/trait.Error.html#method.invalid_value) here.

Reply:
@RReverser, are you suggesting creating a new `ErrorKind`?

Reply:
Missed this. No, I meant using Serde's own method.

Reply:
Thanks. I chose `Error::invalid_value`.